packages/eslint-plugin/src/rules/no-for-in-array.ts

```diff
@@ -33,8 +33,9 @@ export default createRule({
         const type = getConstrainedTypeAtLocation(services, node.right);
 
         if (
-          isTypeArrayTypeOrUnionOfArrayTypes(type, checker) ||
-          (type.flags & ts.TypeFlags.StringLike) !== 0
```

Comment:
I'm not sure if this is necessary, as there's no test that fails if this is removed. TypeScript already reports this as an error, so I don't think the rule has to report this twice ([playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3mgXNjMAligOYDcAUBQGZIzwAUy6WU8x2AlPAN4UC%2BFIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDMPpbmtAIbRoQwuiiJRfSODABfEAqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)).